code/modules/mining/mine_turfs.dm

```diff
@@ -127,20 +126,8 @@
 			if(excavation_amount)
 				to_chat(user, SPAN_NOTICE("You start exacavating [src]."))
 				if(I.use_tool(user, src, WORKTIME_SLOW, tool_type, FAILCHANCE_NORMAL, required_stat = STAT_COG))
-					to_chat(user, SPAN_NOTICE("You finish exacavating [src]."))
-
-					if(excavation_level + excavation_amount >= 100 )
-						//if players have been excavating this turf, leave some rocky debris behind
-						new /obj/structure/boulder
-
+					to_chat(user, SPAN_NOTICE("You finish excavating [src]."))
 					excavation_level += excavation_amount
-
-					//drop some rocks
-					next_rock += excavation_amount * 10
-					while(next_rock > 100)
-						next_rock -= 100
-						var/obj/item/ore/O = new(src)
```

Comment:
I know that this part looks suspicious but actual mineral dropping is handled in other part of code. Before xenoarcheology removal this gave a rock with xenoarcheology data.